pkg/github/github.go

```diff
@@ -93,13 +117,102 @@ func NewClient(ctx context.Context, token string, opts ...Option) *GitHubClient
 	return g
 }
 
+// ListIssues lists all issues and returns their numbers in a repository matching the given criteria.
+func (g *GitHubClient) ListIssues(ctx context.Context, owner, repo string, labels []string, state string) ([]*Issue, error) {
+	backoff := retry.NewFibonacci(g.cfg.initialRetryDelay)
+	backoff = retry.WithMaxRetries(g.cfg.maxRetries, backoff)
+	backoff = retry.WithCappedDuration(g.cfg.maxRetryDelay, backoff)
+
+	var response []*Issue
+
+	if err := retry.Do(ctx, backoff, func(ctx context.Context) error {
+		issues, resp, err := g.client.Issues.ListByRepo(ctx, owner, repo, &github.IssueListByRepoOptions{
+			Labels: labels,
+			State:  state,
+		})
+		if err != nil {
+			if _, ok := ignoredStatusCodes[resp.StatusCode]; !ok {
+				return retry.RetryableError(err)
+			}
+
+			return fmt.Errorf("failed to list issues: %w", err)
+		}
+
+		for _, i := range issues {
+			response = append(response, &Issue{Number: i.Number})
+		}
```

Comment:
There's a race condition here with the retry logic. It's possible that new issues have been created while you're paginating, in which case you'd likely end up with duplicates. It would be better to accumulate all these issues into a map keyed by a unique ID (e.g. issue number) and then convert to a slice once you're done. You could also just return the map.

Reply:
Done